lib/src/component/ref_util.dart

```diff
@@ -52,52 +54,153 @@ Ref<T> createRef<T>() {
 ///
-/// __Example__:
+/// __Example 1:__ Forwarding refs to DOM components
 ///
-///     UiFactory<DomProps> DivForwarded = forwardRef<DomProps>((props, ref) {
-///       return (Dom.div()
-///         ..ref = ref
-///         ..className = 'special-class'
-///       )(
-///         props.children
-///       );
-///     })(Dom.div);
+/// ```dart
+/// import 'dart:html';
+/// import 'package:over_react/over_react.dart';
+/// import 'package:over_react/react_dom.dart' as react_dom;
 ///
-/// ___ OR ___
+/// // ---------- Component Definition ----------
 ///
-///     UiFactory<FooProps> FooForwarded = forwardRef<FooProps>((props, ref) {
-///       return (Foo()
-///         ..forwardedRef = ref
-///       )();
-///     })(Foo);
+/// final FancyButton = forwardRef<DomProps>((props, ref) {
+///   final classes = ClassNameBuilder.fromProps(props)
+///     ..add('FancyButton');
 ///
-///     UiFactory<FooProps> Foo = _$Foo;
+///   return (Dom.button()
+///     ..addProps(getPropsToForward(props, onlyCopyDomProps: true))
+///     ..className = classes.toClassName()
+///     ..ref = ref
+///   )('Click me!');
+/// })(Dom.button);
 ///
-///     mixin FooProps on UiProps {
-///       Ref forwardedRef;
-///     }
+/// // ---------- Component Consumption ----------
 ///
-///     @Component2()
-///     class FooComponent extends UiComponent2<FooProps> {
-///       @override
-///       render() {
-///         return (Dom.button()
-///           ..ref = props.forwardedRef
-///         )('Click this button');
-///       }
-///     }
+/// void main() {
+///   setClientConfiguration();
+///   final ref = createRef<Element>();
+///
+///   react_dom.render(
+///       (FancyButton()
+///         ..ref = ref
+///         ..onClick = (_) {
+///           print(ref.current.outerHtml);
+///         }
+///       )(),
+///       querySelector('#idOfSomeNodeInTheDom')
+///   );
+///
+///   // You can still get a ref directly to the DOM button:
+///   final buttonNode = ref.current;
+/// }
+/// ```
+///
+/// __Example 2:__ Forwarding refs in higher-order components
+///
+/// ```dart
+/// import 'dart:html';
+/// import 'package:over_react/over_react.dart';
+/// import 'package:over_react/react_dom.dart' as react_dom;
+///
+/// // ---------- Component Definitions ----------
+///
+/// final FancyButton = forwardRef<DomProps>((props, ref) {
+///   final classes = ClassNameBuilder.fromProps(props)
+///     ..add('FancyButton');
+///
+///   return (Dom.button()
+///     ..addProps(getPropsToForward(props, onlyCopyDomProps: true))
+///     ..className = classes.toClassName()
+///     ..ref = ref
+///   )('Click me!');
+/// })(Dom.button);
+///
+/// final LogProps = forwardRef<LogPropsProps>((props, ref) {
+///   return (_LogProps()
+///     ..addProps(props)
+///     .._forwardedRef = ref
+///   )('Click me!');
+/// })(_LogProps);
+///
+/// UiFactory<LogPropsProps> _LogProps = _$_LogProps;
+///
+/// mixin LogPropsProps on UiProps {
```

Comment:
Please double check the boilerplate here - I updated this example from what it was previously.